apps/wing/src/commands/test.ts

```diff
@@ -246,16 +247,31 @@ function noCleanUp(synthDir: string) {
   );
 }
 
+function filterTests(tests: string[], regexString?: string): string[] {
+  if (regexString) {
+    const regex = new RegExp(regexString);
+    tests = tests.filter((test) => {
+      // Extract test name from the string
+      // root/env0/test:<testName>
+      const testName = test.split(":").pop();
+      return testName ? regex.test(testName) : false;
+    });
+  }
+
+  return pickOneTestPerEnvironment(tests);
+}
+
 async function testSimulator(synthDir: string, options: TestOptions) {
```

Comment:
The filtering logic needs to happen across all targets, not just the sim. Without code duplication ideally.